gpytorch/lazy/lazy_tensor.py

```diff
@@ -379,6 +379,33 @@ def _approx_diag(self):
         """
         return self.diag()
 
+    @cached(name="cholesky")
+    def _cholesky(self):
+        """
+        (Optional) Cholesky-factorizes the LazyTensor
+
+        ..note::
+            This method is used as an internal helper. Calling this method directly is discouraged.
+
+        Returns:
+            (LazyTensor) Cholesky factor
+        """
+        from .non_lazy_tensor import NonLazyTensor
+        cholesky = psd_safe_cholesky(self.evaluate().double()).to(self.dtype)
+        return NonLazyTensor(cholesky)
+
+    def _cholesky_solve(self, rhs):
+        """
+        (Optional) Assuming that `self` is a Cholesky factor, computes the cholesky solve
+
+        ..note::
+            This method is used as an internal helper. Calling this method directly is discouraged.
+
+        Returns:
+            (LazyTensor) Cholesky factor
+        """
+        return cholesky_solve(rhs.double(), self.evaluate().double()).to(self.dtype)
```

Comment:
Unfortunately, using `cholesky_solve` is dangerous and will prevent us from getting gradients through predictions because it doesn't have a gradient implemented. We might need to manually do this with two `torch.trtrs` solves, which does support gradients.

Reply:
Or better, use `trtrs` when either input requires grad, and use `potrs` otherwise (I do this e.g. for the fantasy cache updates in `ExactPredictionStrategy`)

Reply:
@jacobrgardner the way everything is written right now - `_cholesky_solve` is never called directly, but only as part of the forward/backward methods of the `InvMatmul` function. So we do get a backward pass.

It'll be the same for `InvQuadLogDet` as well.